apps/ewallet/lib/ewallet/web/orchestrator.ex

```diff
@@ -0,0 +1,67 @@
+defmodule EWallet.Web.Orchestrator do
+  @moduledoc """
+  This module should be used in every controller to deal with searching, filtering,
+  preloading and sorting.
+  """
+  alias EWallet.Web.{MatchAllParser, Paginator, Preloader, SearchParser, SortParser}
+
+  def query(query, overlay, attrs \\ nil) do
+    query
+    |> build_query(overlay, attrs)
+    |> Paginator.paginate_attrs(attrs)
+  end
+
+  def build_query(query, overlay, attrs \\ nil) do
+    query
+    |> preload_to_query(overlay, attrs)
+    |> MatchAllParser.to_query(attrs, overlay.filter_fields())
+    |> SearchParser.to_query(attrs, overlay.search_fields, default_mapped_fields())
+    |> SortParser.to_query(attrs, overlay.sort_fields, default_mapped_fields())
+  end
+
+  def all(records, overlay, attrs \\ nil)
+
+  def all(records, _overlay, %{"preload" => preload}) when is_map(preload) do
+    Preloader.preload_all(records, preload)
+  end
+
+  def all(records, _overlay, %{"preload" => preload}) when is_list(preload) do
+    Preloader.preload_all(records, preload)
+  end
+
+  def all(records, overlay, _attrs) do
+    Preloader.preload_all(records, overlay.default_preload_assocs())
+  end
+
+  def one(record, overlay, attrs \\ nil)
```

Comment:
Can we rename to `preload_one`? Reading it the first time, I was thinking if it's similar to [`Repo.one/2`](https://hexdocs.pm/ecto/Ecto.Repo.html#c:one/2), but it's about preloading.